src/lib/thread_pool.c

```diff
@@ -91,12 +91,7 @@ struct tp *tp_start(int num_threads)
 	}
 
 	tp = malloc_or_die(sizeof(struct tp) + int_to_uint(num_threads) * sizeof(pthread_t *));
-
 	tp->num_threads = num_threads;
-	if (num_threads == 0) {
```

Comment:
There's a fix for that on #1564

Reply:
I'll remove that from here then and will review #1564 😝